### ISSUE TITLE
Installing this extension causes the Microsoft Python extension to crash

### ISSUE THREAD

**Issue by DonJayamanne** [Microsoft/vscode-tools-for-ai#18]
Looks like there's something in this extension that causes the [Microsoft Python Extension](https://github.com/Microsoft/vscode-python) to crash.

See here for confirmation and further info:
https://github.com/Microsoft/vscode-python/issues/432#issuecomment-352151831

Would like to resolve this, unfortunately the source is not available. Please let us know how we can help.

**Comment by chris-lauren** [comment]
Thanks for letting us know Don! Haoxiang and team.can someone please look into this?

Don would be great to work more closely together especially as we plan out the next couple months. I'll setup time in the new year. Welcome to Microsoft by the way! I know it's been a little while but we haven't met yet ;)

Get Outlook for Android<https://aka.ms/ghei36>



From: Don Jayamanne
Sent: Friday, December 15, 6:24 PM
Subject: [Microsoft/vscode-tools-for-ai] Installing this extension causes the Microsoft Python extension to crash (#18)
To: Microsoft/vscode-tools-for-ai
Cc: Subscribed


Looks like there's something in this extension that causes the Microsoft Python Extension<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoft%2Fvscode-python&data=04%7C01%7CChris.Lauren%40microsoft.com%7C024a60a2ffcc48ccbc0608d5442c2654%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636489878763925737%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwifQ%3D%3D%7C-1&sdata=fkQgJP%2FDdwNltjIQue0%2BDDTzodrhsRLefZVApEpODeQ%3D&reserved=0> to crash.
See here for confirmation and further info:
Microsoft/vscode-python#432 (comment)<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoft%2Fvscode-python%2Fissues%2F432%23issuecomment-352151831&data=04%7C01%7CChris.Lauren%40microsoft.com%7C024a60a2ffcc48ccbc0608d5442c2654%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636489878763925737%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwifQ%3D%3D%7C-1&sdata=6oDXTFMVo6SIh%2BTqo6d%2BJVmZvUuz34OdW%2BhojMNRNrw%3D&reserved=0>
Would like to resolve this, unfortunately the source is not available. Please let us know how we can help.

**Comment by DonJayamanne** [comment]
Sure thing, just let me know how I can help. For starters I'd  love to try and help with this issue.

**Comment by SShehriyar** [comment]
The errors that I receive which I believe cause the Microsoft Python Extension to crash are
```json 
{
	"activationTimes": {
		"startup": false,
		"codeLoadingTime": 381,
		"activateCallTime": 0,
		"activateResolvedTime": 1889,
		"activationEvent": "onLanguage:python"
	},
	"runtimeErrors": [
		{
			"name": "TypeError",
			"message": "Cannot read property 'log' of undefined"
		},
		{
			"name": "TypeError",
			"message": "Cannot read property 'log' of undefined"
		},
		{
			"name": "TypeError",
			"message": "Cannot read property 'log' of undefined"
		},
		{
			"name": "TypeError",
			"message": "Cannot read property 'log' of undefined"
		}
	]
}
```

**Comment by DonJayamanne** [comment]
Looks like this issue is back. We have updated reflect-metadata to the latest version in the Python extension. 
Please could you also update your extension to the latest version of `reflect-metadata`. 
Here's the link with details on the issue and the fix (fix = update reflect metadata to latest version) https://github.com/inversify/InversifyJS/issues/737

**Comment by linmajia** [comment]
Fixed in 0.1.6 last December.
